carbon-design-system/carbon-addons-iot-react
# 2620 table components in table cell headers are not readable

Closes #2620 

**Summary**
Menu items in the combobox for column filters were not visible in and keyboard users could not take advantage of the hover tooltip. Another problem, that has gotten a separate issue is the truncation of the placeholder text that also leaves the keyboard user without the full text. 

The first issue is solved by allowing the menu items expand beyond the width of the parent combobox. Decided that max-width for the menu should be set by the application if needed.

The second issue is only partly mitigated by removing most of the padding around the controls, hence freeing up more space. See https://github.com/carbon-design-system/carbon-addons-iot-react/issues/2620#issuecomment-910751860

**Change List (commits, features, bugs, etc)**
- Added `menuFitContent` & `horizontalDirection` props &  and functionality to the PAL combobox
- Updated the FilterHeaderRow to us the PAL combobox instead of Carbons
- Removed old CSS that wasn't working and got rid of some complexity
- Wrote unit tests


**Acceptance Test (how to verify the PR)**
1. Go to https://deploy-preview-2797--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-table-statefultable--stateful-example-with-create-save-views
2. Resize the Select column to be really wide and verify that the combobox menu isn't smaller than the combobox itself
3. Resize the Select column to be really thin and verify that the combobox menu expands expands enough to show the menu items without ellipses.
5. Move the Select column to the end of the table and verify that it now expands to the left
6. Check that the above also works in RTL
7. Repeat for story https://deploy-preview-2797--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-table-table--table-example-with-create-save-views
8. Go to story https://deploy-preview-2797--carbon-addons-iot-react.netlify.app?path=/story/1-watson-iot-table-statefultable--simple-stateful-example and enable knobs "hasFilter", "hasResize" and "hasColumnSelection" and repeat step 2-7.

**Regression Test (how to make sure this PR doesn't break old functionality)**
Test the other table stories that have filters and verify that they still work the same except for the above changes.



**Things to look for during review**
